### PR TITLE
Cherry picking RC to main

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Description of the Change:**  All commits to RC are cherry-picked to main. Some changes were already merged to main in #701

### CODE REVIEW

File: .github/workflows/build-wheel-macos-x86_64.yaml [hunk 1]
```diff
@@ -376,7 +376,10 @@ jobs:
 
     - name: Install Catalyst
       run: |
-        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --extra-index-url https://test.pypi.org/simple
+        python${{ matrix.python_version }} -m pip install pennylane-lightning==0.36.0
+        python${{ matrix.python_version }} -m pip install --upgrade git+https://github.com/PennyLaneAI/pennylane@v0.36.0-rc0
+        python${{ matrix.python_version }} -m pip install jax==0.4.23 jaxlib==0.4.23 tomlkit 'scipy<1.13' 'numpy<2' 'diastatic-malt>=2.15.1'
+        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --no-deps
```
Comment (dime10) [inline .github/workflows/build-wheel-macos-x86_64.yaml:382]:
```suggestion
        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --extra-index-url https://test.pypi.org/simple
```

File: .github/workflows/build-wheel-macos-arm64.yaml [hunk 2]
```diff
@@ -406,7 +406,11 @@ jobs:
 
     - name: Install Catalyst
       run: |
-        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --extra-index-url https://test.pypi.org/simple
+        python${{ matrix.python_version }} -m pip install pennylane-lightning==0.36.0
+        python${{ matrix.python_version }} -m pip install --upgrade git+https://github.com/PennyLaneAI/pennylane@v0.36.0-rc0
+        python${{ matrix.python_version }} -m pip install jax==0.4.23 jaxlib==0.4.23 tomlkit 'scipy<1.13' 'numpy<2' 'diastatic-malt>=2.15.1'
+        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --no-deps
+
```
Comment (dime10) [inline .github/workflows/build-wheel-macos-arm64.yaml:413]:
```suggestion
        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --extra-index-url https://test.pypi.org/simple
```

File: .github/workflows/build-wheel-linux-x86_64.yaml [hunk 3]
```diff
@@ -416,7 +417,12 @@ jobs:
 
     - name: Install Catalyst
       run: |
-        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --extra-index-url https://test.pypi.org/simple
+        python${{ matrix.python_version }} -m pip install pennylane-lightning==0.36.0
+        python${{ matrix.python_version }} -m pip install 'pennylane-lightning[kokkos]==0.36.0'
+        python${{ matrix.python_version }} -m pip install --upgrade git+https://github.com/PennyLaneAI/pennylane@v0.36.0-rc0
+        python${{ matrix.python_version }} -m pip install jax==0.4.23 jaxlib==0.4.23 tomlkit 'scipy<1.13' 'numpy<2' 'diastatic-malt>=2.15.1'
+        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --no-deps
+
```
Comment (dime10) [inline .github/workflows/build-wheel-linux-x86_64.yaml:425]: Let's add some todo's next time so it's always clear what is a temporary fix :)
```suggestion
        python${{ matrix.python_version }} -m pip install $GITHUB_WORKSPACE/dist/*.whl --extra-index-url https://test.pypi.org/simple
```

**Review by dime10 — APPROVED** [review]
Thanks, looks solid 🦾

File: .github/workflows/build-wheel-linux-x86_64.yaml [hunk 4]
```diff
@@ -327,6 +327,7 @@ jobs:
           CXX_COMPILER=$(which g++) \
           OQC_BUILD_DIR="$(pwd)/oqc-build" \
           RT_BUILD_DIR="$(pwd)/runtime-build" \
+          PYTHON=$(which python${{ matrix.python_version }}) \
```
[the remark this answers is not in the document]
Comment (erick-xanadu) [reply .github/workflows/build-wheel-linux-x86_64.yaml:330]: https://github.com/PennyLaneAI/catalyst/pull/726/commits/6f5355cd629ef5b11fc9910ff1214d9f7a7c1266 Thanks for the reminder!

**Pull request merged**